plugins/menu_plugin.py

```diff
@@ -0,0 +1,150 @@
+        if servings:
+            return servings
+
+    def make_api_request_for_events(self):
+        query = 'query {events{edges{node{name, action, startDate, endDate}}}}'
+        res = make_api_request(query)
+        events = res.get('events')
+        if events and 'edges' not in events:
+            return events
+
+    def servings_to_dict(self, servings):
+        new_dict = {}
+
+        for obj in servings:
+            meal = obj['menuItem']['meal']['name']
+            menu_item = obj['menuItem']
+            new_obj = {
+                'public_id': obj['publicId'],
+                'course': menu_item['course']['name'],
+                'sequence_order': menu_item['course']['sequenceOrder'],
+                'dish': menu_item['dish']['name']
+            }
+            if new_dict.get(meal):
+                new_dict[meal].append(new_obj)
+            else:
+                new_dict[meal] = [new_obj]
+
+        # Re-order the meals by sequence order
+        return {
+            key: sorted(
+                value,
+                key=itemgetter('sequence_order')
+            ) for key, value in new_dict.items()
+        }
+
+    def get_meals(self, timetable, day):
+        """
+        Get meals for a particular timetable.
+        day is a string either today, tomorrow, yesterday, or any weekday
+        """
+        servings = self.make_api_request_for_servings(
+            timetable, self.date_helper.date_to_str(day)
+        )
+
+        if servings is not None:
+            meals = self.servings_to_dict(servings)
+            return meals
+
+    def get_event(self, day):
+        events = self.make_api_request_for_events()
+        event_list = []
+
+        if events:
+            date = parser.parse(self.date_helper.date_to_str(day)).isoformat()
+            for event in events:
+                start_date = parser.parse(event['startDate']).isoformat()
+                end_date = parser.parse(event['endDate']).isoformat()
+
+                if (date >= start_date) and (date <= end_date):
+                    event_list.append(event)
+        return event_list
+
+    def meals_check_context_update(self, meals, context, day):
+        if meals and not self.get_event(day):
+            context.update({'meals': meals})
+        else:
+            context.update({'no_meals': True})
+
+
+@respond_to('menu', re.IGNORECASE)
+def menu(message):
+    menu_helper = MenuHelper()
+    date_helper = DateHelper()
+    days = date_helper.get_days()
+    # Convert message text to list to remove multiple spaces that may have
+    # been mistakenly added by the user and convert the list back to string
+    message_text_list = message.body['text'].lower().split()
+    len_msg_text_list = len(message_text_list)
+    message_text = ' '.join(message_text_list)
+    timetable_res = TimetableAPIUtils()
+    timetable_names = timetable_res.list_timetable_names()
+
+    if len_msg_text_list > 1:
+        timetable_name = message_text_list[1]
```

Comment:
Not in this case, hence the reason for checking the length of the list of message being sent in by the user. We would use it to check if that timetable given exists.